tests/integration/test_herding_basic_weighted.py

```diff
@@ -61,6 +63,13 @@ def test_herding_basic(self) -> None:
                     herding coreset MMD",
             )
 
+            self.assertLess(
+                mmd_stein,
+                mmd_random,
+                msg="MMD for random sampling was unexpectedly lower than \
+                    Stein thinning coreset MMD",
+            )
```

Comment:
Yes. I've raised a separate ticket to remove the comparison with random from `tests/unit`, since there's a chance that random is better depending on data generation and PNRG settings. There's an argument that they should remain in `tests/integration`, but we haven't decided on a better approach to the above yet.